src/ring_theory/finiteness.lean

```diff
@@ -550,3 +551,75 @@ ring_hom.finite_presentation.of_finite_type
 end finite_presentation
 
 end alg_hom
+
+section monoid_algebra
+
+namespace add_monoid_algebra
+
+open algebra add_submonoid
+
+variables {R : Type*} {M : Type*} [comm_ring R] [add_comm_monoid M]
+
+lemma surjective_of_closure {S : set M} (hS : closure S = ⊤) :
+  function.surjective (mv_polynomial.aeval (λ (s : S), of R M s.1) :
```

Comment:
There's an evil refl-"cast" from `M` to `multiplicative M` happening here that you should probably make explicit:
```suggestion
  function.surjective (mv_polynomial.aeval (λ (s : S), of R M (multiplicative.of_add ↑s)) :
```